lhs.js

```diff
@@ -42,6 +47,33 @@ class Lhs {
         }
     }
 
+    /*
+     *  Further specify type of node you'd like to select. Can be chained with :func:`type` or :func:`dom`.
+     *  Example usage: ``dom('p').when(fnode => fnode.element.id === 'foo')``
+     *  @param {function} function accepts an fnode and returns a boolean.
+     */
+    when(pred) {
+        let lhs = this.clone();
+        lhs._predicate = pred;
+        return lhs;
+    }
+
+    /*
+     * Of all the dom nodes selected by type() or dom(), return only
+     * the fnodes that satisfy all the predicates imposed by calls to
+     * when()
+     */
+    fnodesSatisfyingWhen(fnodes) {
+        const ret = [];
+        const pred = this._predicate;
+        fnodes.forEach(function (fnode) {
+            if (pred(fnode)) {
+                ret.push(fnode);
+            }
+        });
+        return ret;
```

Comment:
Is there a reason not to shorten this to a call to filter, like below?

```js
fnodesSatisfyingWhen(fnodes) {
    return fnodes.filter(this._predicate);
}
```

Reply:
Ah yes forgot about filter, will change this!